lib/core/src/swapper/boltz/mod.rs

```diff
@@ -89,11 +92,33 @@ impl<P: ProxyUrlFetcher> BoltzSwapper<P> {
 
         let boltz_url = boltz_api_base_url.unwrap_or(self.config.default_boltz_url().to_string());
 
-        let boltz_client = self.boltz_client.get_or_init(|| BoltzClient {
-            inner: BoltzApiClientV2::new(boltz_url, Some(CONNECTION_TIMEOUT)),
+        let mut ws_auth_api_key = None;
+        let mut headers = HeaderMap::new();
+        if boltz_url == BREEZ_SWAP_PROXY_URL {
+            match &self.config.breez_api_key {
+                Some(api_key) => {
+                    ws_auth_api_key = Some(api_key.clone());
+                    headers.insert("authorization", format!("Bearer {api_key}").parse()?);
+                }
+                None => {
+                    bail!("Cannot start Boltz client: Breez API key is not set")
```

Comment:
We should support not setting the API key (RE #554). If it's not set we should revert to the Boltz URL like the Breez server currently returns